firmware/application/apps/ui_playlist.cpp

```diff
@@ -54,54 +54,55 @@ void PlaylistView::load_file(std::filesystem::path playlist_path) {
             if ((int)one_char[0] >= ' ') {
                 line += one_char[0];
             } else if (one_char[0] == '\n') {
-                total_tracks++;
                 txtline_process(line);
                 line.clear();
             }
         }
         if (line.length() > 0) {
-            total_tracks++;
             txtline_process(line);
         }
     }
     playlist_masterdb = playlist_db;
-    // text_track.set(to_string_dec_uint(track_number) + "/" + to_string_dec_uint(total_tracks)); //removed because there's no space for it
+    text_track.set(to_string_dec_uint(track_number) + "/" + to_string_dec_uint(total_tracks) + " " + now_play_list_file.string());
     tracks_progressbar.set_max(total_tracks);
     button_play.focus();
     return;
 }
 
 void PlaylistView::txtline_process(std::string& line) {
     playlist_entry new_item;
-    rf::Frequency f;
     size_t previous = 0;
     auto current = std::string::npos;
+    // read freq
     current = line.find(',');
-    std::string freqs = line.substr(0, current);
+    if (current == std::string::npos) return;
+    errno = 0;
+    new_item.replay_frequency = strtoll(line.substr(0, current).c_str(), nullptr, 0);
+    if (new_item.replay_frequency == 0 || errno == EINVAL || errno == ERANGE)
+        return;
+    // read file
     previous = current + 1;
-    current = line.find(',', previous);
-    std::string file = line.substr(previous, current - previous);
+    if ((current = line.find(',', previous)) == std::string::npos) return;
+    new_item.replay_file = "/" + line.substr(previous, current - previous);
+    // read samplerate
     previous = current + 1;
+    errno = 0;
+    new_item.sample_rate = strtoll(line.substr(previous).c_str(), nullptr, 10);
+    if (new_item.sample_rate == 0 || errno == EINVAL || errno == ERANGE)
+        return;
+    // optionnal read delay
     current = line.find(',', previous);
-    uint32_t sample_rate = strtoll(line.substr(previous).c_str(), nullptr, 10);
-    previous = current + 1;
-
-    uint32_t item_delay;
-
-    current = line.find(',', previous);
-    if (current == std::string::npos) {  // compatibility with old PPL grammar
-        item_delay = strtoll(line.substr(previous).c_str(), nullptr, 10);
+    if (current == std::string::npos) {
```

Comment:
Is that actually npos? That doesn’t seem right. It wasn’t npos just above when you found the comma above and I don’t see another find?

Reply:
This would be so much easier if we just broke on lines and then had a split function that returned a list of string views. The FileWrapper might be useful here.

Reply:
Nvm, line 94. It wasn’t “green” in the diff so I missed it.